### DISCUSSION TITLE
Patreon?

### DISCUSSION THREAD

**Discussion by xaelectronics** [nickcoutsos/keymap-editor#62] [General]
This tool has been incredibly useful to me. Is there any way for me to make a donation to support your work?

**Comment by nickcoutsos** [comment]
Hi! I'm really flattered, but I feel weird about accepting donations.

Knowing that the tool has been useful means a great deal. For now you can support me by sharing what difficulties you've had in using it or things you'd like to see. Not everything is easily achievable but the discussions can really help to shape my perspective and guide solution design in the longer term.

Thanks again for the kind words!

**Reply by xaelectronics** [reply]
A few suggestions (should I create a new discussion for this?):

1) When changing a normal &kp to &hm or another hold-tap behavior with two keybindings, the existing keypress should become the second keybinding of &hm instead of the first.

Currently, when changing <&kp z> to &hm, the default becomes <&hm z ?> where ? needs to be defined. This makes z the hold behavior rather than tap. In the majority of circumstances, I would expect a kp -> hm remapping to want the kp as the tap rather than hold.

2) tap dance support would be great

3) I added mouse buttons behaviors based on a zmk fork but the keymap editor doesn't recognize them so I had to edit .keymap directly. Tapdance behaviors show up but the mouse buttons don't. It's not a big deal but would be nice to have. 

4) Direct access via simple text editor to the .keymap file would be great for advanced features that aren't supported by the GUI, so that a user doesn't need to switch back and forth between GUI and external text editor.

5) When changing the name of a macro in the GUI, the label isn't automatically set. Every new macro has a default label of "new_macro" and you need to go into settings to change the label. Ideal behavior would change the label name when the GUI name is changed.

Great work on the tool, it's been very useful. Let me know if you ever change your mind about donations.

**Reply by nickcoutsos** [reply]
Let's do separate issues/discussions, I won't be able to keep up otherwise.
